### ISSUE TITLE
Keyboard handling for popups

### ISSUE THREAD

**Issue by rajdhakategraphy** [exyte/PopupView#81]
Keyboard area is not handled for popups

I'm using floater type to show a popup. (bottom popup version 2 from example)

```
.popup(isPresented: $showForgotPassword, type: .floater(verticalPadding: 0, useSafeAreaInset: false), position: .bottom, dragToDismiss: false, closeOnTap: false, backgroundColor: .black.opacity(0.4)) {
            ForgotPasswordView(forgotPasswordViewModel: ForgotPasswordViewModel(email: email))
        }
```

And this is the result

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-13 at 18 25 34](https://user-images.githubusercontent.com/88770870/195602896-591603fd-2cd7-47b6-9db2-5daf0a982d71.png)

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-13 at 18 25 37](https://user-images.githubusercontent.com/88770870/195602924-ec85db4f-1dcc-476d-b8f1-0d9b4fa8a0af.png)

**Comment by f3dm76** [comment]
Hey @rajdhakategraphy, here is what I got while trying to reproduce your situation
![Simulator Screen Shot - iPhone 14 Pro - 2023-01-10 at 15 03 46](https://user-images.githubusercontent.com/9447630/211495097-8cb9ab78-c278-45a1-8c0d-5613b966a161.png | width=100)
![Simulator Screen Shot - iPhone 14 Pro - 2023-01-10 at 15 04 08](https://user-images.githubusercontent.com/9447630/211495110-9f971367-5d56-41c1-b401-9e4d98841ef7.png | width=100)
Please provide your compilable minimal reproducible code sample so I can help you better. Also please keep in mind that keyboard avoidance policy might be better handled from your app's perspective so I don't accidentally force something that you don't expect on your view

**Comment by SeedsOfLove** [comment]
I have the same problem. Here is my code

```
import SwiftUI
import ExytePopupView

struct TestView: View {
    
    @State var isPresented = false
    @State var text = ""
    
    var body: some View {
        Form {
            
            Section {
                Text("Test Text 1")
                    .onTapGesture {
                        isPresented = true
                    }
                Text("Test Text 2")
                
            }
        }
        .popup(isPresented: $isPresented,
               type: .floater(verticalPadding: 0, useSafeAreaInset: false),
               position: .bottom,
               animation: .spring(),
               closeOnTapOutside: true,
               backgroundColor: .black.opacity(0.3)) {
           
            VStack {
                Text("title")
                    .font(.title2)
                TextField("", text: $text)
                    .frame(height: 36)
                    .padding([.leading, .trailing], 10)
                    .background(Color.gray.opacity(0.3))
                    .cornerRadius(10)
            }
            .padding()
            .padding(.bottom, 30)
            .background(Color.yellow)
        }
        
    }
}


struct TestView_Previews: PreviewProvider {
    static var previews: some View {
        TestView()
            .preferredColorScheme(.light)
    }
}
```
![simulator_screenshot_0C2817CA-AE7F-4A3F-8965-84E788A84E5D](https://user-images.githubusercontent.com/34243682/218609470-928bc36f-ab1f-48fa-8248-fea0cea08441.png)

![simulator_screenshot_757B0D80-6486-4161-9026-CF42865E0318](https://user-images.githubusercontent.com/34243682/218609495-84ace9f6-d990-4d58-912b-cf7db584d2de.png)

**Comment by SeedsOfLove** [comment]
@f3dm76

**Comment by blyscuit** [comment]
@SeedsOfLove I think you are missing `.closeOnTap(false)`
```
Form {

            Section {
                Text("Test Text 1")
                    .onTapGesture {
                        isPresented = true
                    }
                Text("Test Text 2")

            }
        }
        .popup(isPresented: $isPresented) {

            VStack {
                Text("title")
                    .font(.title2)
                TextField("", text: $text)
                    .frame(height: 36)
                    .padding([.leading, .trailing], 10)
                    .background(Color.gray.opacity(0.3))
                    .cornerRadius(10)
            }
            .padding()
            .padding(.bottom, 30)
            .background(Color.yellow)
        } customize: {
            $0
                .type(.floater(verticalPadding: 0, useSafeAreaInset: false))
                .position(.bottom)
                .closeOnTap(false)
                .closeOnTapOutside(true)
                .backgroundColor(.black.opacity(0.3))
        }
```

**Comment by victorazzo** [comment]
I can trigger the keyboard bug by adding .isOpaque(true), which used to be a modifier I no longer need. But the examples above don't even have it, so there is something inconsistent with the keyboard behavior.

```
customize: {
                    $0
                    //.isOpaque(true)
                    .closeOnTapOutside(true)
                    .closeOnTap(false)
                    .backgroundColor(.black.opacity(0.1))
                    .type (.floater(verticalPadding: 0.0, useSafeAreaInset: true))
                    .position(.bottom)
             }
```

**Comment by paulpaulzhang** [comment]
I also encountered the same error. I found that when the keyboard popped up, the content in the popup moved up, but did not completely leave the scope of the keyboard
```swift
.popup(isPresented: $showSendBox) {
            sendBox
        } customize: {
            $0
                .type(.floater(verticalPadding: 0, useSafeAreaInset: false))
                .position(.bottom)
                .closeOnTap(false)
                .closeOnTapOutside(true)
                .backgroundColor(.black.opacity(0.3))
        }
```

**Comment by f3dm76** [comment]
Hey guys, please check out version 2.2.1. I added keyboard avoidance for opaque popups, and I tried to remove the little popup-keyboard overlap for non-opaque ones. Please give a try and let me know if there are still problems. Have a great day all of you!

**Comment by victorazzo** [comment]
I can tell the issue persists with version 2.6.0

**Comment by victorazzo** [comment]
@f3dm76 - I'm sorry, you are correct - my issue is not related to issue #81. The good news is that the fix you just pushed in 2.6.1 in response to #141 fixed my issue as well (that is, using `.useKeyboardSafeArea(false)`). Thanks for the support!
